bin/diagnostics.bash

```diff
@@ -308,14 +324,86 @@ get_config_files() {
     done
 }
 
+run_diagnostics_with_metrics() {
+    TOKEN=$("${here}/ops.bash" kubectl "${cluster}" get secret diagnostics-script-access-token -n thanos -ojsonpath='{.data.token}' | base64 -d)
```

Comment:
I'm unsure to understand why we need a SA token here. Why can't the platform administrator use their Kubernetes API access to proxy into Thanos directly?

I'd prefer the diagnostic script not to need a new token. Instead, it should "just" do whatever it can with the privileges of the user invoking it.

Reply:
I kind of like that an SA and a clear Role would be used, so that it is clear what one intends to do with the access that is granted. Makes it much easier to audit, and therefore to not "trust" but rather **verify**.

I would rather see "run this as an admin plz" like the trend of wget-ting some script and piping it into a sudo shell.

Reply:
I like the idea of using a service account token as well, but I think I'd like us to use the `TokenRequest` API via `kubectl create token` instead of creating a dedicated long-lived token.

Reply:
I preferred using a token as to not require some awkward handling of a `kubectl proxy` background process in the script. Will have to keep the SA and RBAC but can switch to creating a token with a duration with kubectl instead.

Reply:
Would it be possible to use kubectl oidc-login manually to retrieve an id token for the current user? I'm also not a fan of this setup either.

Reply:
I've tested this and it actually works, the only thing is that you have to ensure that oidc-login is used, and that you have to gather all arguments from the target kubeconfig.

```bash
# Would be better to actually resolve the current context and associated user.
readarray -t args <<< "$(yq '.users[0].user.exec.args[]' <kubeconfig-path>)"

# Must be oidc-login here
[[ "${args[0]" == "oidc-login" ]] || exit 1

token="$(kubectl "${args[@]}" | yq '.status.token')"
```

Reply:
What I like about using a dedicated service account token is that we call the API with least amount of permissions required as well make way for the diagnostics to be executed as a machine job. It's kind of similar to AWS IAM roles.

This is just an opinion not a blocker from my side though.

Reply:
> What I like about using a dedicated service account token is that we call the API with least amount of permissions required as well make way for the diagnostics to be executed as a machine job. It's kind of similar to AWS IAM roles.

Sure, I would agree if this script wasn't part of the bin scripts, because we are already platform administrator and if we do a request we should do it as ourself, not via indirection.

If this script was placed somewhere else, to be run in say a Kubernetes job or just standalone then I would want it to to have proper roles setup, and service accounts for each intended runner.
There shouldn't be a global sa that everyone runs with, as the sa is the identity.